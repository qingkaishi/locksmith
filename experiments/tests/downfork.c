#include <pthread.h>

void *doit(void *z) {
  *(int *)z = 1;
}
void dofork(int *y) {
  pthread_t t;
  pthread_create(&t,NULL,doit,(void *)y);
}
void main() {
  int x = 1; // should not count as a guarded-by; not shared
  dofork(&x);
  x = 1; // race
}

