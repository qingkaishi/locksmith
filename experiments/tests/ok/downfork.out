************************* STARTING *************************

shared: &x:downfork.c:11

Warning: Possible data race: &x:downfork.c:11 is not protected!
 references:
  dereference of cast:downfork.c:4 at downfork.c:4
    &x:downfork.c:11 => dofork.y:downfork.c:12
                     => y:downfork.c:6 at dofork downfork.c:12
                     => cast:downfork.c:4 at doit downfork.c:8
  locks acquired:
    <empty>
  in: main at downfork.c:10  -> downfork.c:12

  dereference of &x:downfork.c:11 at downfork.c:13
    &x:downfork.c:11
  locks acquired:
    <empty>
  in: main at downfork.c:10 


*************************** DONE ***************************
